components/motor/gpio/motor_encoder.go

```diff
@@ -372,56 +374,67 @@ func (m *EncodedMotor) rpmMonitorPass(pos, lastPos, now, lastTime int64, rpmDebu
 					desiredRPM /= 2
 				}
 			}
-			m.rpmMonitorPassSetRpmInLock(pos, lastPos, now, lastTime, desiredRPM, rotationsLeft, rpmDebug)
+			m.rpmMonitorPassSetRpmInLock(currentRPM, desiredRPM, rotationsLeft, rpmDebug)
 		}
 	}
 }
 
-func (m *EncodedMotor) rpmMonitorPassSetRpmInLock(pos, lastPos, now, lastTime int64, desiredRPM, rotationsLeft float64, rpmDebug bool) {
-	lastPowerPct := m.state.lastPowerPct
-	rotations := float64(pos-lastPos) / float64(m.cfg.TicksPerRotation)
+func (m *EncodedMotor) computeRPM(pos, lastPos, now, lastTime int64) float64 {
 	minutes := float64(now-lastTime) / (1e9 * 60)
-	currentRPM := rotations / minutes
 	if minutes == 0 {
-		currentRPM = 0
+		return 0.0
 	}
-	m.state.currentRPM = currentRPM
-
-	var newPowerPct float64
+	rotations := float64(pos-lastPos) / float64(m.cfg.TicksPerRotation)
+	return rotations / minutes
+}
 
+func (m *EncodedMotor) computeNewPowerPct(currentRPM, desiredRPM float64) float64 {
+	lastPowerPct := m.state.lastPowerPct
 	if math.Abs(currentRPM) <= 0.001 {
 		if math.Abs(lastPowerPct) < 0.01 {
-			newPowerPct = .01 * float64(sign(desiredRPM))
-		} else {
-			newPowerPct = m.computeRamp(lastPowerPct, lastPowerPct*2)
+			// We began stopped. Set the power to a low setting so we can get started.
+			return .01 * float64(sign(desiredRPM))
 		}
-	} else {
-		dOverC := desiredRPM / currentRPM
-		dOverC = math.Min(dOverC, 2)
-		dOverC = math.Max(dOverC, -2)
+		// We've been putting power to the motor, but it's not moving yet. Try increasing the power
+		// to it, and we'll start moving soon.
+		return m.computeRamp(lastPowerPct, lastPowerPct*2)
+	}
 
-		neededPowerPct := lastPowerPct * dOverC
+	dOverC := desiredRPM / currentRPM
+	dOverC = math.Min(dOverC, 2)
+	dOverC = math.Max(dOverC, -2)
 
-		if !math.Signbit(neededPowerPct) {
-			neededPowerPct = math.Max(neededPowerPct, 0.01)
-			neededPowerPct = math.Min(neededPowerPct, 1)
-		} else {
-			neededPowerPct = math.Min(neededPowerPct, -0.01)
-			neededPowerPct = math.Max(neededPowerPct, -1)
-		}
+	// The last power percent resulted in the last RPM measurement. To get to the desired RPM,
+	// multiply by their ratio.
+	neededPowerPct := lastPowerPct * dOverC
 
-		newPowerPct = m.computeRamp(lastPowerPct, neededPowerPct)
+	if !math.Signbit(neededPowerPct) { // neededPowerPct is positive
```

Comment:
Add comment above: 	// Bound the value of the power between 0.01 or 1 in the positive or negative direction.